#352: BallistaContext::collect() logging is too noisy

jgoday opened the pull request:
# Which issue does this PR close?
Fix #352.

 # Rationale for this change
See #352 (logging too noisy inside BallistaContext::collect() inner loop).

# What changes are included in this PR?
Change info messages for trace inside collect loop.

# Are there any user-facing changes?
N/A

andygrove reviewed (commented):
Thanks @jgoday. With this change we will never see output saying that the query is running unless we have trace logging enabled. Would it be possible to show the message with info logging on the first time we see that the job is running?

jgoday commented:
> Thanks @jgoday. With this change we will never see output saying that the query is running unless we have trace logging enabled. Would it be possible to show the message with info logging on the first time we see that the job is running?

Ok, Is ok if we use a flag (first time a job_id is logged) and define a macro info_or_trace_if like ?

```rust

macro_rules! info_or_trace_if {
    ($first:ident, $($arg:tt)+) => (
        if $first {
            info!($($arg)+);
        } else {
            trace!($($arg)+);
        }
        $first = false;
    )
}
```

andygrove commented:
Perhaps we should just log the status if the status has changed since the last time round the loop?

jgoday commented:
@andygrove I have changed the condition to call info/trace to check the job's prev status.
Should we trace all loop iterations or just info! when the status has changed (so we don't need the info_or_trace_if macro) ?

andygrove commented:
> @andygrove I have changed the condition to call info/trace to check the job's prev status.
> Should we trace all loop iterations or just info! when the status has changed (so we don't need the info_or_trace_if macro) ?

Nice. I think we can just log with `info!` and remove the macro in this case.

jgoday commented:
> > @andygrove I have changed the condition to call info/trace to check the job's prev status.
> > Should we trace all loop iterations or just info! when the status has changed (so we don't need the info_or_trace_if macro) ?
> 
> Nice. I think we can just log with `info!` and remove the macro in this case.

Ok, I have changed it that way.

andygrove commented on ballista/rust/client/src/context.rs:
```diff
@@ -198,14 +200,21 @@ impl BallistaContext {
                 DataFusionError::Internal("Received empty status message".to_owned())
             })?;
             let wait_future = tokio::time::sleep(Duration::from_millis(100));
+            let has_status_change = prev_status.map(|x| x != status).unwrap_or(false);
```
```suggestion
            let has_status_change = prev_status.map(|x| x != status).unwrap_or(true);
```
I think the default should be `true` here. If there was no previous status then the status must have changed?

jgoday replied:
You are right, sorry for the mistake :(

andygrove reviewed (approved):
LGTM. I haven't tested this, but plan on doing so tonight. I will merge once I have confirmed that this works as expected.

Thanks @jgoday!

andygrove commented:
```
[2021-05-27T00:55:04Z INFO  ballista::context] Connecting to Ballista scheduler at http://ballista-scheduler:50050
[2021-05-27T00:55:04Z INFO  ballista::context] Job kO3v67p still queued...
[2021-05-27T00:55:04Z INFO  ballista::context] Job kO3v67p is running...
Query 12 iteration 0 took 5751.5 ms
+------------+-----------------+----------------+
| l_shipmode | high_line_count | low_line_count |
+------------+-----------------+----------------+
| MAIL       | 6202            | 9324           |
| SHIP       | 6200            | 9262           |
+------------+-----------------+----------------+
Query 12 avg time: 5751.51 ms
```

:heart:
